test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs

```diff
@@ -383,5 +384,290 @@ public void HistogramNaN()
+                    }
+
+                    labelString = $"{{key='\\[{string.Join(",", expectedArray)}\\]'}}".ToLower();
+                    break;
+                default:
+                    labelString = $"{{key='{Convert.ToBoolean(value)}'}}".ToLower();
+                    break;
+            }
+
+            Assert.Matches(
+                ("^"
+                    + "# TYPE test_counter counter\n"
+                    + $"test_counter{labelString} 10 \\d+\n"
+                    + "$").Replace('\'', '"'),
+                Encoding.UTF8.GetString(buffer, 0, cursor));
+        }
+
+        [Theory]
+        [InlineData(char.MaxValue)]
+        [InlineData("string")]
+        public void StringTypesSupported(object value)
+        {
+            var buffer = new byte[85000];
+            var metrics = new List<Metric>();
+
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
+
+            var counter = meter.CreateCounter<long>("test_counter");
+            counter.Add(10, new KeyValuePair<string, object>("key", value));
+
+            provider.ForceFlush();
+
+            var cursor = PrometheusSerializer.WriteMetric(buffer, 0, metrics[0]);
+
+            var labelString = $"{{key='{Convert.ToString(value)}'}}";
+
+            Assert.Matches(
+                ("^"
+                    + "# TYPE test_counter counter\n"
+                    + $"test_counter{labelString} 10 \\d+\n"
+                    + "$").Replace('\'', '"'),
+                Encoding.UTF8.GetString(buffer, 0, cursor));
+        }
+
+        [Fact]
+        public void StringArrayTypesSupported()
+        {
+            var buffer = new byte[85000];
+            var metrics = new List<Metric>();
+
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
+
+            var charArray = new char[] { 'a', 'b', 'c' };
+            var stringArray = new string[] { "a", "b", "c", string.Empty, null };
+            var tags = new KeyValuePair<string, object>[]
+            {
+                new KeyValuePair<string, object>("charArray", charArray),
+                new KeyValuePair<string, object>("stringArray", stringArray),
+            };
+
+            var counter = meter.CreateCounter<long>("test_counter");
+            counter.Add(10, tags);
+
+            provider.ForceFlush();
+
+            var cursor = PrometheusSerializer.WriteMetric(buffer, 0, metrics[0]);
+
+            var labelString = @"{charArray=""\[\\""a\\"",\\""b\\"",\\""c\\""\]"",stringArray=""\[\\""a\\"",\\""b\\"",\\""c\\"",\\""\\"",null\]""}";
```

Comment:
This looks regex kinda heinous but it's just matching a JSON representation of the string array which includes an empty string and null value.

1. Is Prometheus ok with JSONed arrays?
2. Is Prometheus ok with empty string values in an array?
3. For null values, is the literal `null` ok?

Reply:
https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#comments-help-text-and-type-information

> label_value can be any sequence of UTF-8 characters, but the backslash (\), double-quote ("), and line feed (\n) characters have to be escaped as \\, \", and \n, respectively.